HW1/P4/P4_bfs.py

```diff
@@ -0,0 +1,31 @@
+__author__ = 'xiaowen'
+
+def bfs(node, graph):
+    # initialize graph with node as vertice and initialize distance
+    bfs_rdd = graph.map(lambda (x, y): (x, (0, y)) if (x == node) else (x, (float("inf"), y)))
+    distances = bfs_rdd.map(lambda (x, y): (x, y[0])).collectAsMap()
+    count_nodes = 0
+    counter = graph.context.accumulator(0)
+    while counter.value == 0:
+        bfs_rdd = bfs_rdd.map(lambda x: update_distance(x, distances))
+        distances = bfs_rdd.map(lambda (n, (d, nb)): (n, d)).collectAsMap()
```

Comment:
Problem 4: Not sure though how effective the call to collectAsMap() would be time/space effective.